app/models/member.rb

```diff
@@ -121,6 +121,26 @@ def user_has_namespace_maintainer_access?(user, namespace)
                        access_level: Member::AccessLevel::MAINTAINER)
       end
     end
+
+    def membership_source(namespace, user)
+      if Member.exists?(
+        namespace:, user:
+      )
+        { label: I18n.t('activerecord.models.member.direct') }
+      else
+        source_namespaces = if namespace.project_namespace?
```

Comment:
and in the else, we can just use `member.namespace` instead of having to search for the namespace since we already have it in the `member` model.

Reply:
Updated in 0379505201abd0d8b0a317f46c1819e1c82f5c3e